atomic_reactor/plugins/post_tag_from_config.py

```diff
@@ -43,7 +44,38 @@ class TagFromConfigPlugin(PostBuildPlugin):
 
     TAGS_FILENAME = 'additional-tags'
 
-    def get_tags(self):
+    def __init__(self, tasker, workflow, tags=None):
+        super(TagFromConfigPlugin, self).__init__(tasker, workflow)
+        self.tags = tags
+
+    def parse_and_add_tags(self):
+        class MyFormatter(string.Formatter):
+            """
+            using this because str.format can't handle keys with dots and dashes
+            which are included in some of the labels, such as
+            'authoritative-source-url', 'com.redhat.component', etc
+            """
+            def get_field(self, field_name, args, kwargs):
+                return (self.get_value(field_name, args, kwargs), field_name)
+
+        tags = []
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow,
+                           env_replace=True).labels
+
+        for tag in self.tags.get('unique', []):
+            self.log.debug('Using additional unique tag %s', tag)
+            self.workflow.tag_conf.add_unique_image(tag)
```

Comment:
My comment is not to substitute values in the tag provided, but to actually add a prefix to it.
This also applies to primary tags.